eclair-core/src/main/scala/fr/acinq/eclair/db/pg/PgChannelsDb.scala

```diff
@@ -99,6 +99,11 @@ class PgChannelsDb(implicit ds: DataSource, lock: PgLock) extends ChannelsDb wit
         )(logger)
       }
 
+      def migration78(statement: Statement): Unit = {
+        statement.executeUpdate("CREATE INDEX created_timestamp_idx ON local.channels(created_timestamp)")
+        statement.executeUpdate("CREATE INDEX closed_timestamp_idx ON local.channels(closed_timestamp)")
```

Comment:
> * make the `remoteNodeId` field mandatory when listing closed channels -> this guarantees that the set of rows affected won't ever become extremely large, and we already have an index on `remote_node_id` to make that query efficiently

The whole reason to have this RPC to find out which peer closed a channel. It will be unusable with  mandatory `remoteNodeId`.

I made it optional though. It definitely can be useful.

Reply:
I don't understand, this seems to contradict what you were saying before? If you're reacting to a `ChannelClosed` notification, that notification already contains the `remoteNodeId` (or at least it should).

If you have not received any `ChannelClosed` notification and you're just curious to know if some channels were recently closed without specifying a `remoteNodeId`, this is doable, but will be slightly inefficient because we don't have a DB index (but this isn't an important enough feature to justify adding an index). But if that works well enough for you, I'm ok with it!

Reply:
There's no contradiction. The PR description says that the main purpose of this PR is getting access to the historical data. 

I've been using this RPC on my private build for a long time. Typically I use this RPC for reporting on a regular basis, and I use it for getting data about channels closed by the peers, but just occasionally.

I don't need filtering by remoteNodeId at all, but you need it. That's a good compromise. But, honestly, I would drop it, because without filtering the PR is much cleaner and neat. The users can filter the list any way they want, like I do.

Reply:
But you're talking from the point of view of a small node with a low number of closed channels. What happens when you have more than 100 000 closed channels? What about 1 000 000? You really need to do some filtering once you reach those numbers, that are very reasonable for any LSP.

Reply:
If we really care about 1m+ closed channels, we have to address two major performance killers in the current database design.

1. Boolean conditions in `WHERE` clauses like `is_closed=0` make the DBMS use sequential scan all the time regardless of having any indexes. It affects `channels` RPC in the first place: the DBMS has to iterate over the whole table to find all open channels.
2. For pagination to be consistent we need to sort the rows deterministically. But we don't have relevant indexes, so the DBMS sorts the rows on every query instead of doing index scan.

The fix for the latter issue is obvious. But for the former it's more complicated: `local_channles` table should be split into two tables one for open channels, and the other for closed ones. Each table should have it own set of indexes.
In this case it will handle 1m+ rows much better.

Reply:
Since it's a historical data RPC, the most natural way to limit the number of rows returned is mandatory `--from` and `--to` parameters.

Reply:
> `local_channels` table should be split into two tables one for open channels, and the other for closed ones. 

That was my conclusion as well, that would be much better. But you're right, this is probably an early optimization that isn't necessary yet, we can do that later.